### PR TITLE
Support new TigerBeetle functionality

### PR DESCRIPTION
**Pull request by 50U10FCA7**
Requires #24

# Problem

[`query_accounts`](https://docs.tigerbeetle.com/reference/requests/query_accounts) and [`query_transfers`](https://docs.tigerbeetle.com/reference/requests/query_transfers) TigerBeetle requests (and [`QueryFilter`](https://docs.tigerbeetle.com/reference/query-filter/)) not yet supported.

# Solution

Implement:
- [`QueryFilter`](https://docs.tigerbeetle.com/reference/query-filter) record;
- [`query_accounts`](https://docs.tigerbeetle.com/reference/requests/query_accounts) request;
- [`query_transfers`](https://docs.tigerbeetle.com/reference/requests/query_transfers) request.

### CODE REVIEW

**Comment by 50U10FCA7** [comment]
### FCM
```markdown
Support new TigerBeetle functionality (#26)

- add `QueryFilter`
- impl `Client::query_accounts` and `Client::query_transfers`
```

**Review by tyranron — CHANGES_REQUESTED** [review]

File: core/src/query_filter.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    }
+    pub const fn with_flags(mut self, flags: Flags) -> Self {
+        self.0.flags = flags.bits();
+        self
+    }
+}
+
+impl fmt::Debug for QueryFilter {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("AccountFilter")
```
[the remark this answers is not in the document]
Comment (50U10FCA7) [reply core/src/query_filter.rs]: Typo

File: core/src/handle.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ impl<'a, U> ClientHandle<'a, U>
 where
     U: UserDataPtr,
 {
-    pub fn packet(self, user_data: U, operation: packet::Operation) -> Packet<'a, U> {
+    pub fn packet(self, user_data: U, operation: impl Into<packet::Operation>) -> Packet<'a, U> {
```
[the remark this answers is not in the document]
Comment (50U10FCA7) [reply core/src/handle.rs]: @tyranron #34

**Review by tyranron — APPROVED** [review]
@50U10FCA7 LGTM, thanks! 🍻

@zetanumbers the code changes are relatively simple and mostly copy-pasted from the existing sources. Please, take a look, and if you don't have any objections, I'll merge in 2 days or so.

**Pull request merged**